jira/client.py

```diff
@@ -92,7 +92,8 @@ def emit(self, record):
 #    "which means that there is a big change of having problems. " \
 #    "Possible workaround http://stackoverflow.com/a/17628350/99834" % encoding)
 
-logging.getLogger('jira').addHandler(NullHandler())
+logger = logging.getLogger('jira')
```

Comment:
Interesting. You are right with the `__name__` (I've confused it with func/class name).

> we need to be conscious that this logger name becomes part of our public API and a change to its name is a breaking change

If mapped to `__name__`, then unless top package name changes, it will be stable. `jira.client` is fine for a logger, messages are propagated. However, I guess that its not needed to divide logger namespaces for each module in the package.

Here is a test of propagation:
```
import logging

parent = logging.getLogger('jira')
child = logging.getLogger('jira.client')
parent.addHandler(logging.StreamHandler())
assert child.propagate
assert child.parent is parent
child.error('Test')
```